run-tests.js

```diff
@@ -146,9 +162,12 @@ const sequentialMap = async (input, fn) => {
 const testExchange = async (exchange) => {
 
 /*  Run tests for all/selected languages (in parallel)     */
-
-    const args = [exchange, ...symbol === 'all' ? [] : symbol]
-        , allTests = [
+    let args = [exchange];
+    if (symbol !== undefined && symbol !== 'all') {
+        args.push(symbol);
+    }
+    args = args.concat(exchangeOptions)
```

Comment:
lets concentrate here.
the first item in `args` is `exchange` (as it was before) and that's correct.
but the above `if` clause makes the inclusion of next item (symbol) conditional.  so, the second `args` might be or not `symbol`. if not, then the second item in `args` might be an item from `exchangeOptions`.
however here:
https://github.com/ccxt/ccxt/blob/master/js/test/test.js#L5
as you see the second argument (actually 4th) is expected to be `exchangeSymbol`. 

so, this approach might also lead to misbehavior , as it seems

Reply:
@ttodua but the flags are excluded from that so they don't count, as you can see at the end of the line `process.argv.filter ((x) => !x.startsWith ('--'));`

Reply:
ah, ignore my above msg. `exchangeOptions` expect all of keys to start with `--` which invalidates my above statement (though, still not a robust argvs parsing inside test.js, but as is..)

Reply:
wait, seems not correct. namely, it's "correct for JS" but not for other langs. i.e.  https://github.com/ccxt/ccxt/blob/master/php/test/test_async.php#L60

Reply:
same bug might be triggered in py too, havent checked

Reply:
btw, ArgumentParser removes `--` automatically and you can directly use `args.privateOnly` (instead of i.e. `args.__privateOnly`) right? https://github.com/ccxt/ccxt/blob/master/python/ccxt/test/test_async.py#L399

Reply:
found out about argumentParser, ignore last comment.

Reply:
> wait, seems not correct. namely, it's "correct for JS" but not for other langs. i.e. https://github.com/ccxt/ccxt/blob/master/php/test/test_async.php#L60

@ttodua I didn't add these features to PHP yet

Reply:
> same bug might be triggered in py too, havent checked

@ttodua I think that it will work fine in python because we use an argument parser there 

```
python python/ccxt/test/test_async.py bybit "BTC/USDT" --sandbox          
/Users/cjg/Git/ccxt10/ccxt /Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support
['python/ccxt/test/test_async.py', 'bybit', 'BTC/USDT', '--sandbox']
SYMBOL: BTC/USDT
CODE: BTC
Testing fetch_ticker:BTC/USDT
bybit BTC/USDT ticker 2023-02-01T16:54:00.998Z high: 18102.86 low: 17867.63 bid: 18102.85 ask: 18102.86 volume: 5709310.09180846
Testing fetch_tickers:BTC/USDT
bybit fetched all 205 tickers
Testing fetch_ohlcv:BTC/USDT
bybit fetched 10 OHLCVs
Testing fetch_order_book:BTC/USDT
bybit BTC/USDT order book 2023-02-01T16:54:25.651Z bid: 18102.85 bidVolume: 0.145245 ask: 18102.86 askVolume: 16.924971
Testing fetch_trades:BTC/USDT
bybit BTC/USDT fetched 100 trades
```